DataFormats/MuonReco/src/MuonSelectors.cc

```diff
@@ -901,6 +906,21 @@ bool muon::isHighPtMuon(const reco::Muon& muon, const reco::Vertex& vtx){
 
 }
 
+bool muon::isTrackerHighPtMuon(const reco::Muon& muon, const reco::Vertex& vtx){
+  bool muID =   muon.isTrackerMuon() && muon.track().isNonnull() && (muon.numberOfMatchedStations() > 1);
+  if(!muID) return false;
+
+  bool hits = muon.innerTrack()->hitPattern().trackerLayersWithMeasurement() > 5 &&
+    muon.innerTrack()->hitPattern().numberOfValidPixelHits() > 0; 
+
+  bool momQuality = muon.tunePMuonBestTrack()->ptError()/muon.tunePMuonBestTrack()->pt() < 0.3;
+
+  bool ip = fabs(muon.innerTrack()->dxy(vtx.position())) < 0.2 && fabs(muon.innerTrack()->dz(vtx.position())) < 0.5;
```

Comment:
copy-pasted implementation is usually not a good idea.
If the selections after ```if(!muID) return false``` are supposed to be the same from "first principles", better define a common method .

Reply:
That's how it's defined at the moment and it can change in future, i.e. there is nothing that requires that two selectors to be different by just a few minor points - they have different use cases (Z' vs boosted Z).